test/chainsaw/testmetrics/dynatrace-provider/mock-server.yaml

```diff
@@ -0,0 +1,149 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: mockserver
+spec:
+  ports:
+    - name: serviceport
+      port: 1080
+      protocol: TCP
+      targetPort: serviceport
+  selector:
+    app: mockserver
+  sessionAffinity: None
+  type: ClusterIP
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: mockserver
+  name: mockserver
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: mockserver
+  template:
+    metadata:
+      labels:
+        app: mockserver
+      name: mockserver
+    spec:
+      automountServiceAccountToken: false
+      containers:
+        - env:
+            - name: MOCKSERVER_LOG_LEVEL
+              value: INFO
+            - name: SERVER_PORT
+              value: "1080"
+          image: mockserver/mockserver:mockserver-5.13.0
+          imagePullPolicy: IfNotPresent
+          resources:
+            limits:
+              cpu: 500m
+              memory: 128Mi
+            requests:
+              cpu: 10m
+              memory: 64Mi
+          livenessProbe:
+            failureThreshold: 10
+            initialDelaySeconds: 10
+            periodSeconds: 5
+            successThreshold: 1
+            tcpSocket:
+              port: serviceport
+            timeoutSeconds: 1
+          name: mockserver
+          ports:
+            - containerPort: 1080
+              name: serviceport
+              protocol: TCP
+          readinessProbe:
+            failureThreshold: 10
+            initialDelaySeconds: 2
+            periodSeconds: 2
+            successThreshold: 1
+            tcpSocket:
+              port: serviceport
+            timeoutSeconds: 1
+          volumeMounts:
+            - mountPath: /config
+              name: config-volume
+            - mountPath: /libs
+              name: libs-volume
+      terminationGracePeriodSeconds: 30
+      volumes:
+        - configMap:
+            defaultMode: 420
+            name: mockserver-config
+            optional: true
+          name: config-volume
+        - configMap:
+            defaultMode: 420
+            name: mockserver-config
+            optional: true
+          name: libs-volume
+---
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: mockserver-config
+data:
+  initializerJson.json: |-
+    [
+      {
+        "httpRequest": {
+          "method" : "GET",
+          "path" : "/api/v2/metrics/query",
+          "queryStringParameters" : {
+            "metricSelector" : [ "query" ]
+          }
+          "headers" : {
+            "content-length" : [ "0" ],
+            "connection" : [ "keep-alive" ],
+            "accept-encoding" : [ "gzip,deflate" ],
+            "User-Agent" : [ "Go-http-client/1.1" ],
+            "Authorization" : [ "Api-Token token: mytoken" ]
+          },
```

Comment:
let's get rid of the headers bit entirely, the request and config seems still not to match